float32 torque1
float32 torque2